shared/keystore/deposit_input.go

```diff
@@ -0,0 +1,35 @@
+package keystore
+
+import (
+	"github.com/gogo/protobuf/proto"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// DepositInput for a given key. This input data can be used to when making a
+// validator deposit. The input data includes a proof of possession field
+// signed by the deposit key.
+func DepositInput(depositKey *Key, withdrawalKey *Key) *pb.DepositInput {
+	di := &pb.DepositInput{
+		Pubkey:                      depositKey.PublicKey.Marshal(),
+		WithdrawalCredentialsHash32: withdrawalCredentialsHash(withdrawalKey),
+	}
+
+	// #nosec G104 -- This can only throw if di is nil so lets ignore the error.
+	b, _ := proto.Marshal(di)
+	di.ProofOfPossession = depositKey.SecretKey.Sign(b, params.BeaconConfig().DomainDeposit).Marshal()
```

Comment:
@prestonvanloon 

### Submit deposit

In phase 0, all incoming validator deposits originate from the Ethereum 1.0 PoW chain. Deposits are made to the [deposit contract](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#ethereum-10-deposit-contract) located at `DEPOSIT_CONTRACT_ADDRESS`.

To submit a deposit:

* Pack the validator's [initialization parameters](#initialization) into `deposit_input`, a [`DepositInput`](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#depositinput) SSZ object.
* Set `deposit_input.proof_of_possession = EMPTY_SIGNATURE`.
* Let `proof_of_possession` be the result of `bls_sign` of the `hash_tree_root(deposit_input)` with `domain=DOMAIN_DEPOSIT`.
* Set `deposit_input.proof_of_possession = proof_of_possession`.
* Let `amount` be the amount in Gwei to be deposited by the validator where `MIN_DEPOSIT_AMOUNT <= amount <= MAX_DEPOSIT_AMOUNT`.
* Send a transaction on the Ethereum 1.0 chain to `DEPOSIT_CONTRACT_ADDRESS` executing `deposit` along with `serialize(deposit_input)` as the singular `bytes` input along with a deposit `amount` in Gwei.

_Note_: Deposits made for the same `pubkey` are treated as for the same validator. A singular `Validator` will be added to `state.validator_registry` with each additional deposit amount added to the validator's balance. A validator can only be activated when total deposits for the validator pubkey meet or exceed `MAX_DEPOSIT_AMOUNT`.

Reply:
Thanks! Was this from the validator spec? I was looking at the main spec and didn't see any of this

Reply:
Yep from the latest honest validator spec